tests/5.0/teams_loop/test_target_teams_loop_defaultmap.c

```diff
@@ -0,0 +1,238 @@
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+
+#define N 1024
+
+/**
+  This is a basic test to demonstrate defaultmap(tofrom)
+  clause with target teams loop construct.
+*/
+int testDefaultMapToFrom() {
+  int errors = 0;
+  int device_data[N], host_data[N];
+  for (int i = 0; i < N; i++) {
+    device_data[i] = i;
+    host_data[i] = 0;
+  }
+#pragma omp target teams loop defaultmap(tofrom)
+  for (int i = 0; i < N; i++) {
+    host_data[i] = device_data[i];
+    device_data[i] = 2*i;
+  }
+  // checking results
+  for (int i = 0; i < N; i++) {
+    OMPVV_TEST_AND_SET(errors, host_data[i] != i);
+    OMPVV_TEST_AND_SET(errors, device_data[i] != 2*i);
+  }
+  return errors;
+}
+
+/**
+  This is a basic test to demonstrate defaultmap(from)
+  clause with target teams loop construct.
+*/
+int testDefaultMapFrom() {
+  int errors = 0;
+  int device_data[N], host_data[N];
+  for (int i = 0; i < N; i++) {
+    device_data[i] = i;
+    host_data[i] = 0;
+  }
+#pragma omp target teams loop defaultmap(from)
+  for (int i = 0; i < N; i++) {
+    host_data[i] = i;
+  }
+  // checking results
+  for (int i = 0; i < N; i++) {
+    OMPVV_TEST_AND_SET(errors, host_data[i] != i);
+  }
+  return errors;
+}
+
+/**
+  This is a basic test to demonstrate defaultmap(to)
+  clause with target teams loop construct.
+*/
+int testDefaultMapTo() {
+  int errors = 0;
+  int device_data[N], host_data[N];
+  for (int i = 0; i < N; i++) {
+    device_data[i] = i;
+    host_data[i] = 0;
+  }
+#pragma omp target teams loop defaultmap(to)
+  for (int i = 0; i < N; i++) {
+    host_data[i] = device_data[i];
+  }
+  // checking results
+  for (int i = 0; i < N; i++) {
+    OMPVV_TEST_AND_SET(errors, host_data[i] != 0);
```

Comment:
On a shared memory system or host fallback host_data will be updated (which is correct). This will lead to incorrect error reporting.

Better to do a map(from: host_data) on the target and check for updated host_data.

Reply:
The intent of using "defaultmap(to)" is to ensure that host side memory(process's stack memory) "host_data" is not updated. Hence the check "host_data[i] != 0" at line 71.
Because host_data is host memory allocated in stack, the expectation is that it doesn't come under "shared memory" aspects... Please let us know if your opinion differs...

Reply:
> Because host_data is host memory allocated in stack, the expectation is that it doesn't come under "shared memory" aspects... Please let us know if your opinion differs…

I don't understand the above comment; `host_data` on the host will be updated if on a shared memory system regardless of whether `host_data` is in the stack memory or global memory. 
I think that a better way to handle this issue is to print a warning message if this test runs on the shared memory system or host fallback occurs.

Reply:
added map(from: host_data) as suggested...

Reply:
Change `map(from: host_data)` to `map(tofrom: host_data)` since `host_data` is read first in the target region.
Change  `OMPVV_TEST_AND_SET(errors, host_data[i] != 0);` to `OMPVV_TEST_AND_SET(errors, host_data[i] != i);`

Reply:
One remaining issue is that this test may not necessarily check the behavior of the `defaultmap(to)` clause.

Reply:
@lthakur007, you can potentially add a constant based on omp_is_initial_device() to the host_data to differentiate between host_fallback vs. target execution.